src/actor/mod.rs

```diff
@@ -24,6 +30,17 @@ impl<T: Clone + Send + Sync + 'static> ActorHandle<T> {
     handle
   }
 
+  pub fn spawn_with_shutdown(
```

Comment:
Spawn && SpawnWithShutdown should be merged to one. Why have 2? The default trait impl of ShutdownHook can be no-op and actors can all get an implicit impl of it.